File: app/dexguard/DexGuard-9.2.7/samples/basic/ThreatCastTamperDetection/src/com/example/HelloWorldActivity.java
/*
 * Sample application to illustrate tamper detection with DexGuard.
 *
 * Copyright (c) 2012-2019 Guardsquare NV
 */
package com.example;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.widget.Toast;


public class HelloWorldActivity extends Activity
{
    @Override
    public void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);

        myToast("hello world", this);
        myToast(this, "hello world");
        myToast(1, this);
    }


    private void myToast(String s, Context context)
    {
        Toast.makeText(this, s, Toast.LENGTH_LONG).show();
    }

    private void myToast(Context context, String s)
    {
        Toast.makeText(this, s, Toast.LENGTH_LONG).show();
    }

    private void myToast(int i, Context context)
    {
        Toast.makeText(this, "i: " + i, Toast.LENGTH_LONG).show();
    }
}

File: app/dexguard/DexGuard-9.2.7/samples/advanced/Moshi/src/com/example/HelloWorldActivity.kt
/*
 * Sample application to illustrate processing with DexGuard.
 *
 * Copyright (c) 2012-2020 Guardsquare NV
 */
package com.example

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.TextView
import com.squareup.moshi.Moshi

/**
 * Sample activity that displays "Hello world!".
 */
class HelloWorldActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val response = Response("a", "my_val", "c")
        val moshi: Moshi = Moshi.Builder().build()

        // Make sure serialization is working
        val gson = moshi.adapter(Response::class.java).toJson(response)
        val textView = findViewById<TextView>(R.id.GsonTextView)
        textView.text = gson

        // Make sure deserialization is working
        val adapter = moshi.adapter(Response::class.java)
        val response2 = adapter.fromJson(gson)
    }
}
